migration.go

```diff
@@ -0,0 +1,22 @@
+package config
+
+import "github.com/libp2p/go-libp2p-core/peer"
+
+const DefaultMigrationKeep = "cache"
+
+var DefaultMigrationDownloadSources = []string{"HTTPS", "IPFS"}
+
+// Migration configures how migrations are downloaded and if the downloads are
+// added to IPFS locally
+type Migration struct {
+	// Sources in order of preference where "HTTPS" means our gateways and
+	// "IPFS" means over IPFS. Any other values are interpretes as hostnames
+	// for custom gateways. An empty list means "do the default thing"
+	DownloadSources []string
+	// Whether or not to keep the migration after downloading it.
+	// Options are "discard", "cache", "pin".  Empty string for default.
+	Keep string
+	// Peers lists the nodes to attempt to connect with when downloading
+	// migrations.
+	Peers []peer.AddrInfo
```

Comment:
@Stebalien this field was not in your initial proposal, WDYT?

It seems like this could be nice if we knew some peers that might have the migration binaries. However, it could also get messy (e.g. is the temp node still connecting to the bootstrappers, or just these peers? does the empty list mean "no predetermined peers/only MDNS" or to append to some default list?, etc.).

Reply:
The reason I felt compelled to add it, is because it can take a very long time to find the migrations.  If I configure a peer using this config, then I can get migrations to complete very fast.

A user may have a node that they bring up when only they are going to up when they update/migrate all the nodes on their network, so it is not one of their normal peers.  They want migrations to be found at that node on their local network.

Reply:
@lidel 
>For cases where finding peers with data is important, migrations should be reusing bootstrappers and Peering map from the original config.

The problem is that config must be read _before_ migration is done, and in cases where config needs to be migrated, the config file may not be readable yet.  The `Migration` section of the conifg is special in that it must not require first being migrated for it to be readable.  So that is a "safe" place to put peers to use during migration.

Reply:
I think it's probably safe to _try_ to read the bootstrap list here. Regardless, I think we can punt this in this version because it's unclear what we should.

Reply:
Removed `Peers` from migration config.  IPFS daemon will try to read bootstrapping and peers from existing config, but skip if error reading.